include/exec/__detail/__system_context_replaceability_api.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2025 Lucian Radu Teodorescu, Lewis Baker
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef STDEXEC_SYSTEM_CONTEXT_REPLACEABILITY_API_H
+#define STDEXEC_SYSTEM_CONTEXT_REPLACEABILITY_API_H
+
+#include "stdexec/__detail/__execution_fwd.hpp"
+
+#include <typeindex>
+
+
+/// Implementation-defined mechanism of querying a system context interface of type `id`.
+extern void* __query_system_context_interface(std::type_index id) noexcept;
```

Comment:
On the usage of `std::type_index`.

Fair point. Please note that this is an QOI issue, and not something we want to specify in the proposal. I could imagine implementations that would use simple enums here.
I will change our implementation to reflect the schema with UUIDs.

Reply:
On constness:

If the replaceability API is with C++ classes, then `const` doesn't make sense. Operations exposed on the interface may not be `const`.
If we are moving back to a C-like interface, then it probably makes sense to have it `const`.

Reply:
> On constness

I agree - if the pattern is to return a struct containing function-pointers then returning a pointer-to-const makes sense. If on the other hand, you are returning a pointer to an object with virtual functions, some of which are logically non-const, then I agree it does not makes sense to return pointer-to-const here.

> I could imagine implementations that would use simple enums here.

One of the use-cases I can imagine people may want to support is the ability to add implementation-specific or user-defined extensions. e.g. to be able to query an interface onto the system context that gives access to qt-specific, or libdispatch-specific facilities, or possibly to query interfaces onto algorithm-specific optimization interfaces. e.g. to allow algorithms to dispatch to an optimized implementations instead of some default implementation in terms of schedule() or bulk().